metals/src/main/scala/scala/meta/internal/metals/DefinitionProvider.scala

```diff
@@ -53,6 +55,30 @@ final class DefinitionProvider(
     }
   }
 
+  def symbolOccurrences(
+      params: TextDocumentPositionParams
+  ): java.util.List[DocumentHighlight] = {
+    val source = params.getTextDocument.getUri.toAbsolutePath
+    val result = semanticdbs.textDocument(source)
+    result.documentIncludingStale
+      .flatMap { doc =>
+        val ResolvedSymbolOccurrence(_, symbolOccurrence) =
+          positionOccurrence(source, params, doc)
+        symbolOccurrence.map { occ =>
+          val allOccurrences = doc.occurrences
+            .filter { cur =>
+              cur.symbol == occ.symbol
+            }
+            .flatMap(_.range)
+            .map { range =>
+              new DocumentHighlight(range.toLSP)
```

Comment:
This range may be outdated, we will need to use `TokenEditDistance` to adjust the tokens to the current buffer.

Reply:
See `ReferenceProvider` for an example